### PR TITLE
refactor: decouple skill display score from system.score (#13)

### LINKED ISSUE #13: Decouple skill display score from system.score (sheet-prep mutation)
## Background

`_prepareCharacterItems` (`src/module/actor/actor-sheet.ts:107`) writes a *display* score back to `i.system.score` on the live `Item` document on every sheet render — `base + mod + attribute`. Other call sites consistently treat `system.score` as `base + mod` and add the attribute themselves, so they double-count the attribute after a sheet render until the next `actor.update()` triggers a `prepareDerivedData()` reset.

PR #12 made the mutation idempotent (it stops *compounding* across re-renders) but did not eliminate the underlying conflict. Direct skill-roll clicks happen to dodge it because the roll dialog reads `data-score` from the DOM, which was rendered from the inflated value at template time. The indirect roll paths still double-count.

### Affected readers

| Reader | Treats `system.score` as | After sheet render | Correct? |
|---|---|---|---|
| `data-score` on `.rolldialog` (template) | Final score | `base+mod+attr` | ✓ |
| `item.ts:84` `getScore()` | `base+mod`, adds attr | `base+mod+2×attr` | ✗ |
| `roll-setup.ts:347, 363` melee/brawl | `base+mod`, adds attr | `base+mod+2×attr` | ✗ |
| `item.ts:259, 273` weapon → skill chain | `base+mod`, adds attr | `base+mod+2×attr` | ✗ |
| `roll-execute.ts:276, 285, 302, 317` `rollMin` | `base+mod`, adds attr | `base+mod+2×attr` | ✗ |

## Approach

Promote `system.total` to the canonical display source and stop mutating `system.score`.

`prepare-data.ts:89-96` already populates `i.system.total` with the display value, but almost nothing reads it. Routing templates through `system.total` lets `system.score` stay clean as `base + mod` while `system.total` is the "ready to roll" display value. The pure helper added in #12 (`actor-helpers/skill-score.ts`) becomes the shared implementation between `prepare-data.ts` and any sheet-side computation.

### Phase 1 — Foundation
- [ ] Audit `actor-helpers/prepare-data.ts:89-96` for completeness across skill / specialization, advanced flag, metaphysics attribute, flat-skills mode.
- [ ] Replace its inline math with `computeSkillDisplayScore()` so `prepare-data` and any future helper share one implementation.
- [ ] Extend `tests/domain/skill-score.test.ts` (or add a parallel domain test) to cover the branches `prepare-data.ts` exercises.

### Phase 2 — Templates (mechanical)
Switch ~30 references from `system.score` to `system.total` in display contexts (visible dice + `data-score` attributes). Keep references in roll-setup helpers untouched — they want `base + mod` and add attribute themselves.

| Template | Estimated refs |
|---|---|
| `src/templates/actor/common/tabs/attribute-column.html` | 4 |
| `src/templates/actor/character/tabs/attributes.html` | 6 |
| `src/templates/actor/character/tabs/metaphysics.html` | 5 |
| `src/templates/actor/character/create-skill-column.html` | 8 |
| `src/templates/actor/character/create-character-skills.html` | ~5 |
| `src/templates/metaphysicsRoll.html` | 4 |

### Phase 3 — Drop the mutation
- [ ] Remove the `i.system.score = computeSkillDisplayScore(...)` call from `_prepareCharacterItems`. Both the skill and specialization branches.
- [ ] Drop the `typeof i.system.attribute !== "undefined"` guard introduced in #12 (no longer needed once we're not computing).

### Phase 4 — Roll-path audit
After Phase 3, `system.score` is consistently `base + mod` again. Verify (no code change expected, just confirmation):
- [ ] `item.ts:84` `getScore()` — `score + attr` is correct
- [ ] `roll-setup.ts:347, 363` melee/brawl — same
- [ ] `item.ts:259, 273` weapon → skill chain — same
- [ ] `roll-execute.ts:276, 285, 302, 317` `rollMin` — same

Add a domain-style test that simulates "render then roll without intervening update" and asserts the resulting score is correct.

### Phase 5 — Documentation
- [ ] Doc comment on `system.total` in the actor schema explaining its semantics: *display value, includes the linked attribute; consumers building roll formulas should use `system.score` (=`base+mod`) and add the attribute themselves.*
- [ ] CHANGELOG note on the field-path change for downstream module authors.

## Risks

- **Third-party modules / Babele packs reading `system.score` for display** — they'll continue to get a valid `base+mod` value, just not the combined dice. Document the migration path in the CHANGELOG and call out `system.total` as the new field.
- **Edge case in `prepare-data.ts`** — if any branch path was missed (e.g. a creature actor with no `attributes`), `system.total` could be undefined and templates render `0D+0`. Domain tests for every actor type's skill rendering mitigate.

## Effort

~3–4 hours: 30 min Phase 1, ~1.5 h Phase 2 (mechanical but careful), 15 min Phase 3, 30 min Phase 4 audit, 30 min tests + docs.

## Branch / PR

- Branch: `fix/skill-score-display-decouple`
- Single focused PR
- Pre-merge smoke: every sheet mode (Normal / Advance / Free-Edit) × {melee, brawl, ranged, direct skill click} × {standard, advanced} × {`flat_skills` on, `flat_skills` off}.

## Origin

Spotted by Copilot on PR #12 ([review comment](https://github.com/Jan-Ka/foundryvtt-opend6-space/pull/12#discussion_r3150770928)). PR #12 reduced the symptom (compounding) but did not remove the root cause (mutation of live document data conflicting with downstream readers' assumed semantics).

### PR DESCRIPTION
**Pull request by Jan-Ka**
## Summary
Resolves the long-standing conflict between sheet-render mutation of `i.system.score` and roll-side consumers that treat `system.score` as `base + mod` and add the linked attribute themselves. After this PR:

- `system.score` is **always** `base + mod` (own progression).
- `system.total` is the **display value** (`base + mod + attribute`, respecting flat-skills + advanced-skill rules), populated by `prepareDerivedActorData()`.
- `_prepareCharacterItems` no longer mutates `system.score`.

Closes #13.

## Phased commits
- `8d9a57f` — **Phase 1**: route `prepare-data.ts` through `computeSkillDisplayScore()` (handles flatSkills + advanced skill correctly; spec gets non-advanced treatment).
- `b37ddf4` — **Phase 2**: codemod 43 actor-side template refs from `system.score` → `system.total`. Item edit sheets unchanged. Codemod script committed at `scripts/codemod-system-score-to-total.mjs`.
- `376d27d` — **Phase 3**: drop the `i.system.score = ...` mutation in `_prepareCharacterItems` for both skill and spec branches.
- `08ec7d2` — **Phase 5**: docstring on `skill-fields.ts` describing the score/total convention.

Phase 4 (roll-path audit) was code-change-free — every consumer already treats `system.score` as `base + mod` and adds the attribute itself. Verified at:
- `src/module/item/item.ts:84` `getScore()` and weapon→skill chain
- `src/module/apps/roll-helpers/roll-setup.ts` (skill, spec, melee/brawl, weapon)
- `src/module/apps/roll-helpers/roll-execute.ts` rollMin sites
- `src/module/system/utilities/skills.ts` `getScoreFromSkill`

## ⚠️ Needs gameplay smoke before merge
This decouples display from roll-formula values, so verifying both is essential. The full matrix from the issue:

- [ ] **Sheet modes**: Normal / Advance / Free-Edit
- [ ] **Roll triggers**: melee, brawl, ranged, direct skill click
- [ ] **Skill kinds**: standard skill, advanced skill (`(A) ...`), specialization
- [ ] **Settings toggles**: `flat_skills` on, `flat_skills` off
- [ ] **Regression check**: render the sheet, then roll the same skill from the dialog without an intervening actor update — produced score should match the displayed dice (the bug this PR closes)
- [ ] **Metaphysics tab**: `metaphysicsRoll.html` dialog populates dice/pips correctly
- [ ] **Character creation**: `create-skill-column` and `create-character-skills` views show the right dice
- [ ] **Item sheet**: opening a skill or specialization item directly shows the raw `base + mod` (unchanged behavior)

## Migration / compatibility note
Third-party modules or Babele packs reading `system.score` for display will continue to get a valid `base + mod` value — they just won't get the combined dice. Field path for the combined display value is now `system.total`.

## Other checks
- [x] `pnpm run check` — lint + typecheck + 162 unit + 250 domain tests pass
- [x] No new domain test failures from removing the mutation

### CODE REVIEW

File: src/module/actor/actor-sheet.ts [hunk 1]
```diff
@@ -124,33 +123,8 @@ export class OD6SActorSheet extends HandlebarsApplicationMixin(ActorSheetV2) {
...
-                if (typeof i.system.attribute !== "undefined") {
-                    i.system.score = computeSkillDisplayScore({
-                        base: i.system.base,
-                        mod: i.system.mod,
-                        attributeScore:
-                            actorData.system.attributes?.[i.system.attribute.toLowerCase()]?.score,
-                        flatSkills: OD6S.flatSkills,
-                    });
-                }
                 specializations.push(i);
```
[the remark this answers is not in the document]
Comment (Jan-Ka) [reply src/module/actor/actor-sheet.ts:128]: Confirmed — Copilot is right, the same mutation still lives in `_prepareVehicleItems` (lines 175-176, 181-182) and `_prepareStarshipItems` (lines 215-216, 221-222). PR #13 phase 3 was scoped to character sheets only, so vehicle/starship double-counting persists when sheets re-render against an already-prepared item.

Tracking as a follow-up rather than expanding this PR — opening a separate issue to extend the `system.total` migration to vehicle/starship sheets (mirrors what was done for characters).

[the remark this answers is not in the document]
Comment (Jan-Ka) [reply src/module/actor/actor-sheet.ts:128]: Tracked as #29.

File: src/module/data/item/fields/skill-fields.ts [hunk 2]
```diff
@@ -1,5 +1,23 @@
 const fields = foundry.data.fields;
 
+/**
+ * Skill / specialization shared progression fields.
+ *
+ * Score-field convention (consumers must follow):
+ *   - `base + mod`     → own progression in pips. Persisted; reset by
+ *                        `applyMods()`.
+ *   - `score`          → derived `base + mod`. Roll-formula consumers read
+ *                        this and add the linked attribute themselves.
```
[the remark this answers is not in the document]
Comment (Jan-Ka) [reply src/module/data/item/fields/skill-fields.ts]: Reworded in 9f324e4. Split the bullet so it's explicit that `base` and `mod` are the persisted inputs (with `base` carrying any persisted advances), and `score` is the derived `base + mod` recomputed by `applyMods()` — not an independent input.

Used a slightly different wording from the suggestion to avoid duplicating the `system.total` paragraph below, but the substance matches.

**Comment by Jan-Ka** [comment]
## Manual smoke results

Smoked against a live Foundry v14 world (od6s v2.1.2). All headline + decoupling assertions pass; bundled a small sheet-scroll CSS fix that surfaced while exercising the item edit sheet.

### Setup

```js
(async () => {
  const old = game.actors.filter(x => x.name?.startsWith('smoke-')).map(x => x.id);
  if (old.length) await Actor.deleteDocuments(old);

  const actor = await Actor.create({
    name: 'smoke-pc', type: 'character',
    system: { attributes: {
      agi: { base: 12, mod: 0 }, kno: { base: 6, mod: 0 },
      mec: { base: 6, mod: 0 }, per: { base: 6, mod: 0 },
      str: { base: 6, mod: 0 }, tec: { base: 6, mod: 0 },
    }},
  });

  await actor.createEmbeddedDocuments('Item', [
    { name: 'Pistol',  type: 'skill',          system: { base: 6, mod: 0, attribute: 'agi' } },
    { name: 'Stealth', type: 'skill',          system: { base: 9, mod: 0, attribute: 'agi', isAdvancedSkill: true } },
    { name: 'Sniping', type: 'specialization', system: { base: 12, mod: 0, attribute: 'agi', skill: 'Pistol' } },
  ]);
  await actor.sheet.render(true);
  await new Promise(r => setTimeout(r, 200));
})();
```

### Headline regression check (the bug #13 fixes)

Score stays raw, total carries the combined display value, `_prepareCharacterItems` no longer mutates `system.score`.

```js
const a = game.actors.find(a => a.name === 'smoke-pc');
console.log({
  pistol:  { score: a.items.find(i => i.name === 'Pistol').system.score,  total: a.items.find(i => i.name === 'Pistol').system.total },
  stealth: { score: a.items.find(i => i.name === 'Stealth').system.score, total: a.items.find(i => i.name === 'Stealth').system.total },
  sniping: { score: a.items.find(i => i.name === 'Sniping').system.score, total: a.items.find(i => i.name === 'Sniping').system.total },
});
```

**Live result:**
```json
{
  "pistol":  { "score": 6,  "total": 18 },
  "stealth": { "score": 9,  "total": 9  },
  "sniping": { "score": 12, "total": 24 }
}
```

✅ `score` stays at `base + mod`. `total` is `base + mod + attribute` (or just `base + mod` for advanced skills).

### Roll formulas (no double-add)

The mutation bug also caused rolls to double-count the attribute. Verify:

| Item | Expected total dice | Live formula | Total |
|---|---|---|---|
| Pistol (score 6 + agi 12 = 18 pips → 6D) | 6 dice | `5d6[Base] + 1dwx6[Wild]` | 21 |
| Sniping (score 12 + agi 12 = 24 pips → 8D) | 8 dice | `7d6[Base] + 1dwx6[Wild]` | 37 |

✅ Wild die replaces one base, so 5d6+wild = 6D, 7d6+wild = 8D. Attribute counted once.

### `flat_skills` toggle

```js
await game.settings.set('od6s', 'flat_skills', true);
actor.prepareData();  // sheet.render() doesn't re-prepare; explicit call needed
// inspect totals…
await game.settings.set('od6s', 'flat_skills', false);
actor.prepareData();
```

**Live result with `flat_skills: true`:**
- Pistol: `score: 6, total: 6`
- Stealth: `score: 9, total: 9`
- Sniping: `score: 12, total: 12`

✅ `total` collapses to raw progression (no attribute).

### Item edit sheet (raw progression)

Opening the Pistol item directly: Attributes tab shows **Dice: 2, Pips: 0, Modifier: 0** — i.e. 6 raw pips, displayed as 2D. No inflation from the actor-side combined value.

✅ Item edit form unchanged.

### Sheet scroll fix (bundled)

Item edit sheet did not scroll on smaller viewports — pre-existing layout bug, same shape as the settings-dialog fix in PR #24. Extended `.od6s.settings-config` rule to also match `.od6s.sheet` (covers both actor and item sheets). Committed as `88c97ad`.

Verified post-fix:
- Item sheet now scrolls internally on smaller viewports across all three tabs (Description / Attributes / Labels). ✅
- Actor sheet renders unchanged after the rule widened. ✅

### Findings

1. **Bug #13 is dead.** `system.score` is never mutated by `_prepareCharacterItems`. `system.total` is the display path. No double-add in rolls.
2. **Phase 4 (roll-path audit) holds in practice.** Every consumer continues to add the attribute exactly once.
3. **Pre-existing sheet scroll bug fixed.** Bundled here because it surfaced while exercising the item-edit case from the PR's smoke matrix.
4. **Useful smoke pattern.** `actor.sheet.render(true)` does not re-run `prepareDerivedActorData` — explicit `actor.prepareData()` is required to observe setting toggles. Worth folding into the runbook.
